### PR TITLE
Drop async lib from environment unit test

### PR DESCRIPTION
**Pull request by mrickard**
Replaced async lib with async functions
Replaced fs with fs/promises
Replaced rimraf with fs/promises

Signed-off-by: mrickard <maurice@mauricerickard.com>

## Proposed Release Notes
In environment unit test: 
* Replaced async lib with async functions
* Replaced fs with fs/promises
* Replaced rimraf with fs/promises

## Links
https://issues.newrelic.com/browse/NR-39257

### CODE REVIEW

File: test/unit/environment.test.js [hunk 1]
```diff
@@ -259,11 +257,43 @@ describe('the environment scraper', function () {
...
+        .mkdir(dirp)
+        .then(() => null)
+        .catch((err) => {
+          if (err.code !== 'EEXIST') {
+            return err
+          }
+          return null
+        })
+      // vestigial--mkdirDb is the async callback, invisibly appended to arguments when apply is used
+      if (mkdirDb) {
```
Comment (mrickard) [inline test/unit/environment.test.js:271]: I could see removing this conditional block entirely, along with the `mkdirDb` property.

File: test/unit/environment.test.js [hunk 2]
```diff
@@ -259,11 +257,38 @@ describe('the environment scraper', function () {
 
   describe('with symlinks', function () {
     const nmod = path.resolve(__dirname, '../helpers/node_modules')
+    const makeDir = (dirp) => {
```
Comment (jmartin4563) [inline test/unit/environment.test.js:260]: It seems weird to use both async/await and then/catch style, do you think we could update this function to be async/await for consistency?

Comment (mrickard) [reply test/unit/environment.test.js:260]: It should be something I can update--this already converted callback-style `fs` to `fs/promises`, which I could wrap in a try/catch block. Let's see.

Comment (mrickard) [reply test/unit/environment.test.js:260]: Should be updated!

**Review by jmartin4563 — APPROVED** [review]

**Pull request merged**